### DISCUSSION TITLE
Documentation?

### DISCUSSION THREAD

**Discussion by phil-s** [eclipse-oct/open-collaboration-tools#42] [General]
I may have missed something here, but... where's the documentation for the protocols?

(I.e. for people who are curious about what would be involved in supporting this for some currently-unsupported text editor.)

I've looked at:

- the readme at https://github.com/TypeFox/open-collaboration-tools
- the announcement at https://www.typefox.io/blog/open-collaboration-tools-announcement/
- https://www.open-collab.tools/
- the readme at https://github.com/TypeFox/open-collaboration-tools/tree/main/packages/open-collaboration-protocol

The very high-level description of "The Open Collaboration Protocol" in the announcement is the nearest thing I could find.

**Comment by msujew** [comment]
Hey @phil-s,

thanks for the question. The documentation for the protocol is currently work in progress while the initial reference implementation (i.e. this repo) is still in flux. We'll soon (i.e. likely in a few weeks) start publishing the protocol.

Aside from that, to get a grip on what needs to be done to support the protocol, please see our reference implementation (split across the `open-collaboration-protocol`, `open-collaboration-yjs` and `open-collaboration-vscode` packages).

**Comment by spoenemann** [comment]
There's also an early concept document here: https://github.com/eclipse-theia/theia/blob/93e9e802c37846d6232e0a76b6c6c7bb09403726/collaboration-concept.md

But it hasn't been updated since last year.

**Comment by diegobes** [comment]
Any update on the protocol documentation ?
I want to explore implementing my own server side and not relay on the reference implementatio .
Thanks

**Reply by msujew** [reply]
See also https://github.com/eclipse-oct/open-collaboration-tools/issues/78.
